### PR TITLE
Pages 1 part

### PR DESCRIPTION
**Pull request by MaxDaniLOVE**
update previous pr

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -5,6 +5,7 @@
     "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
     "no-console": "off",
     "no-use-before-define": 0,
-    "no-param-reassign": 1
+    "no-param-reassign": 1,
+    "prettier/prettier": "off"
```
Comment (maximsan) [inline .eslintrc.json:9]: nonono, you disable all prettier rules, it does not solve the issue

File: src/UI/Buttons/DangerButton.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const DangerButton = ({ onClick, children }) => {
+  return (
+    <button onClick={onClick} className='btn btn-danger' type='button'>
+      {children}
+    </button>
```
Comment (maximsan) [inline src/UI/Buttons/DangerButton.js:8]: Reuse you Button component, just change className

File: src/UI/Buttons/GoBackButton.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const GoBackButton = ({ onClick }) => {
+  return (
+    <button onClick={onClick} className='btn btn-return' type='button'>
```
Comment (maximsan) [inline src/UI/Buttons/GoBackButton.js:6]: Reuse you Button component, just change className

File: src/UI/Buttons/LinkButton.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import PropTypes from 'prop-types';
+
+const LinkButton = ({ children, link }) => {
+  return (
+    <button className='btn' type='button'>
```
Comment (maximsan) [inline src/UI/Buttons/LinkButton.js:7]: Reuse you Button component

File: src/UI/Buttons/OutlineButton.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const OutlineButton = ({ onClick, children }) => {
+  return (
+    <button onClick={onClick} className='btn btn-outline' type='button'>
```
Comment (maximsan) [inline src/UI/Buttons/OutlineButton.js:6]: Reuse you Button component, just change className

File: src/UI/Buttons/SuccessButton.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const SuccessButton = ({ onClick, children }) => {
+  return (
+    <button onClick={onClick} className='btn btn-success' type='button'>
```
Comment (maximsan) [inline src/UI/Buttons/SuccessButton.js:6]: Reuse you Button component, just change className

File: src/UI/Modal/Modal.js [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+        <div className='modal__content'>{children}</div>
+        <div className='modal__footer'>
+          {isDetailMode ? null : <SuccessButton onClick={onSubmit}>Save</SuccessButton>}
+          <GoBackButton onClick={onModalClose} />
+        </div>
+      </form>
+    </div>
+  );
+  return showModal ? createPortal(content, modalDomElement) : null;
+};
```
Comment (maximsan) [inline src/UI/Modal/Modal.js:23]: Try to write component as in docs

File: src/UI/Modal/Modal.scss [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  width: 100vw;
+  background-color: #0000002a;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  .modal {
+    border-radius: 1rem;
+    padding: 1rem;
+    height: fit-content;
+    width: 420px;
```
Comment (maximsan) [inline src/UI/Modal/Modal.scss:15]: use responsive approach - max-width ...

File: src/UI/Status/Status.scss [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+}
+.active {
+  color: $charcoal;
+}
+.failed,
+.success,
+.active {
+  font-weight: 400;
+  font-size: 1.2rem;
+}
```
Comment (maximsan) [inline src/UI/Status/Status.scss:16]: name file in camelCase

File: src/UI/Modal/Modal.scss [hunk 10]
```diff
@@ -0,0 +1,24 @@
...
+    background-color: $baby-blue;
+    display: inherit;
+    flex-direction: column;
+    &__footer {
+      display: flex;
+      justify-content: space-between;
+    }
+  }
+}
```
Comment (maximsan) [inline src/UI/Modal/Modal.scss:24]: name file in camelCase

File: src/components/MembersPageModal/MembersPageModal.js [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    }
+    return (
+      <div className='form-inputs' key={id}>
+        <label htmlFor={id}>
+          {label}
+          <input
+            value={isEditMode ? inputPlaceholder : undefined}
+            type={type}
+            id={id}
+            onChange={(e) => onFormChange(e)}
```
Comment (maximsan) [inline src/components/MembersPageModal/MembersPageModal.js:40]: create method

File: src/components/MembersPageModal/MembersPageModal.js [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+          id={id}
+          type={type}
+          options={options}
+          onChange={onFormChange}
+          data={registerData}
+          isEditMode={isEditMode}
+        />
+      );
+    }
+    const covertationDict = {
```
Comment (maximsan) [inline src/components/MembersPageModal/MembersPageModal.js:25]: convertation ?

File: src/components/MembersPageModal/MembersPageModal.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+      <h3>{title}</h3>
+      <div className='modal__content_container'>{inputsLabels}</div>
+    </>
+  );
+};
+
+MembersPageModal.propTypes = {
+  onFormChange: PropTypes.func.isRequired,
+  isEditMode: PropTypes.bool.isRequired,
+  registerData: PropTypes.objectOf(PropTypes.any).isRequired,
```
Comment (maximsan) [inline src/components/MembersPageModal/MembersPageModal.js:59]: not any

File: src/components/MembersTasksTable/MembersTasksTable.js [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+          <TableHeader headers={headers} />
+          <tbody>{membersTasksTableBody}</tbody>
+        </>
+      </Table>
+    </Layout>
+  );
+};
+
+MembersTasksTable.propTypes = {
+  userTasks: PropTypes.arrayOf(PropTypes.any).isRequired,
```
Comment (maximsan) [inline src/components/MembersTasksTable/MembersTasksTable.js:49]: not any

File: src/components/MembersTasksTable/MembersTasksTable.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+        </td>
+        <td className='admin-btns'>
+          <SuccessButton>Success!</SuccessButton>
+          <DangerButton>Fail!</DangerButton>
+        </td>
+      </tr>
+    );
+  });
+  return (
+    <Layout className='table-wrapper'>
```
Comment (maximsan) [inline src/components/MembersTasksTable/MembersTasksTable.js:37]: I guess your layout already has the same className or is it another one ?

File: src/pages/MembersProgressPage.js [hunk 16]
```diff
@@ -0,0 +1,76 @@
...
+        });
+      });
+    }
+  }
+
+  sortFromOldToNew = (progress) => {
+    const sorted = [...progress];
+    sorted.sort((a, b) => (a.trackDate > b.trackDate ? 1 : -1));
+    return sorted;
+  };
```
Comment (maximsan) [inline src/pages/MembersProgressPage.js:53]: helper method

File: src/pages/MembersProgressPage.js [hunk 17]
```diff
@@ -0,0 +1,76 @@
...
+
+  render() {
+    const { progress, isLoaded, memberName } = this.state;
+    return (
+      <div className='table-wrapper'>
```
Comment (maximsan) [inline src/pages/MembersProgressPage.js:58]: Layout ?

File: src/utils/inputsParser.js [hunk 18]
```diff
@@ -0,0 +1,13 @@
+import { stringToDate } from './convertDate';
+
+const inputsParser = (value, id, data) => {
```
Comment (maximsan) [inline src/utils/inputsParser.js:3]: inputsParser is not correct

File: src/services/Firebase.js [hunk 19]
```diff
@@ -0,0 +1,99 @@
...
+
+  deleteUser = async (id) => {
+    // TODO add deliting users tasks and progress
+    try {
+      await this.database
+        .collection('users')
+        .doc(id)
+        .delete();
+    } catch (error) {
+      throw new Error("Can't delete user data. Try later.");
```
Comment (maximsan) [inline src/services/Firebase.js:96]: Do you catch it anywhere ?

File: src/services/Firebase.js [hunk 20]
```diff
@@ -0,0 +1,99 @@
...
+      throw new Error("Can't get user data. Try later.");
+    }
+  };
+
+  getUsersData = async () => {
+    try {
+      const usersData = await this.database.collection('users').get();
+      return usersData;
+    } catch (error) {
+      throw new Error("Can't get users data. Try later.");
```
Comment (maximsan) [inline src/services/Firebase.js:35]: Do you catch it anywhere ?

File: src/pages/MembersTasksPage.js [hunk 21]
```diff
@@ -0,0 +1,69 @@
...
+          memberName: name,
+        });
+      });
+    }
+  }
+
+  render() {
+    const { userTasks, isLoaded, memberName } = this.state;
+    return (
+      <div className='table-wrapper'>
```
Comment (maximsan) [inline src/pages/MembersTasksPage.js:51]: Layout ?

File: .eslintrc.json (hunk 1 above)
Comment (MaxDaniLOVE) [reply .eslintrc.json:9]: done

File: src/UI/Buttons/DangerButton.js (hunk 2 above)
Comment (MaxDaniLOVE) [reply src/UI/Buttons/DangerButton.js:8]: done

File: src/UI/Buttons/GoBackButton.js (hunk 3 above)
Comment (MaxDaniLOVE) [reply src/UI/Buttons/GoBackButton.js:6]: done

File: src/UI/Buttons/LinkButton.js (hunk 4 above)
Comment (MaxDaniLOVE) [reply src/UI/Buttons/LinkButton.js:7]: done

File: src/UI/Buttons/OutlineButton.js (hunk 5 above)
Comment (MaxDaniLOVE) [reply src/UI/Buttons/OutlineButton.js:6]: done

File: src/UI/Buttons/SuccessButton.js (hunk 6 above)
Comment (MaxDaniLOVE) [reply src/UI/Buttons/SuccessButton.js:6]: done

File: src/components/MembersTasksTable/MembersTasksTable.js (hunk 15 above)
Comment (MaxDaniLOVE) [reply src/components/MembersTasksTable/MembersTasksTable.js:37]: done

File: src/pages/MembersProgressPage.js (hunk 17 above)
Comment (MaxDaniLOVE) [reply src/pages/MembersProgressPage.js:58]: done

File: src/pages/MembersTasksPage.js (hunk 21 above)
Comment (MaxDaniLOVE) [reply src/pages/MembersTasksPage.js:51]: done

File: src/components/MembersPageModal/MembersPageModal.js (hunk 12 above)
Comment (MaxDaniLOVE) [reply src/components/MembersPageModal/MembersPageModal.js:25]: renamed to convertationDict

File: src/components/MembersPageModal/MembersPageModal.js (hunk 13 above)
Comment (MaxDaniLOVE) [reply src/components/MembersPageModal/MembersPageModal.js:59]: done

File: src/components/MembersTasksTable/MembersTasksTable.js (hunk 14 above)
Comment (MaxDaniLOVE) [reply src/components/MembersTasksTable/MembersTasksTable.js:49]: done

File: src/UI/Status/Status.scss (hunk 9 above)
Comment (MaxDaniLOVE) [reply src/UI/Status/Status.scss:16]: done

File: src/UI/Modal/Modal.scss (hunk 10 above)
Comment (MaxDaniLOVE) [reply src/UI/Modal/Modal.scss:24]: done

File: src/UI/Modal/Modal.scss (hunk 8 above)
Comment (MaxDaniLOVE) [reply src/UI/Modal/Modal.scss:15]: done

File: src/components/MembersPageModal/MembersPageModal.js (hunk 11 above)
Comment (MaxDaniLOVE) [reply src/components/MembersPageModal/MembersPageModal.js:40]: already done

File: src/pages/MembersProgressPage.js (hunk 16 above)
Comment (MaxDaniLOVE) [reply src/pages/MembersProgressPage.js:53]: done

File: src/services/Firebase.js (hunk 19 above)
Comment (MaxDaniLOVE) [reply src/services/Firebase.js:96]: done

File: src/services/Firebase.js (hunk 20 above)
Comment (MaxDaniLOVE) [reply src/services/Firebase.js:35]: done

File: src/utils/inputsParser.js (hunk 18 above)
Comment (MaxDaniLOVE) [reply src/utils/inputsParser.js:3]: done

File: src/UI/Modal/Modal.js (hunk 7 above)
Comment (MaxDaniLOVE) [reply src/UI/Modal/Modal.js]: rewrite

File: src/UI/Modal/Modal.js [hunk 22]
```diff
@@ -7,20 +7,49 @@ import './modal.scss';
...
+    if (showModal) {
+      modalDomElement.appendChild(this.el);
+    }
+  }
+
+  componentDidUpdate(prevProps) {
+    const { showModal } = this.props;
+    if (prevProps.showModal === showModal) {
+      return null;
+    }
```
Comment (maximsan) [inline src/UI/Modal/Modal.js:28]: it's seems redundant, you already return null in the end

File: src/UI/Modal/Modal.js [hunk 23]
```diff
@@ -7,20 +7,49 @@ import './modal.scss';
...
+    if (showModal) {
+      modalDomElement.appendChild(this.el);
+    } else {
+      modalDomElement.removeChild(this.el);
+    }
+    return null;
+  }
+
```
Comment (maximsan) [inline src/UI/Modal/Modal.js:36]: Did you forget about componentWillUnmount?

**Pull request merged**